Admin site functionality to delete a site

apburnes opened the pull request:
The UI implementation to close out #2835 which allows admins to delete a site from the admin site page.

## Open questions to get your opinions
- Should the admin api see or search for soft deleted sites?
- If so, do we want to return the deleted site page?

## Acceptance Criteria
**Admin Client**
- [x] Adds a site delete form to the site page
- [x] Adds a verification modal to confirm you want to delete the site
- [x] Sends a delete method to `/site/{id}` to delete a site

**Backend API**
- [x] Add a route to the admin sites 
- [x] Add a controller to the admin sites route
- [x] Test delete request

apburnes commented:
I've got a couple open questions about UX that we can clarify, but its ready to go for the most part.

davemcorwin commented on api/admin/controllers/site.js:
```diff
@@ -69,4 +73,25 @@ module.exports = wrapHandlers({
+        await ProxyDataSync.removeSite(site);
+      }
+
+      await site.destroy();
```
Do all of the steps here need to happen serially, or could they happen concurrently?

davemcorwin replied:
also, this logic is identical to the logic in the user client, perhaps it should be refactored to a common service...

apburnes replied:
👍 Pulled out a `SiteDestroy` service

davemcorwin commented on api/admin/controllers/site.js:
```diff
@@ -69,4 +73,25 @@ module.exports = wrapHandlers({
+      await site.destroy();
+      return res.json(serializeNew(site, true));
+    } catch (error) {
+      logger.error([`site@id=${site.id}`, error.message, error.stack].join('\n\n'));
```
I'm not a huge fan of putting these messages in the logs since no one ever sees them. We need to find a better way of surfacing when these things happen...

apburnes replied:
Put a `To Do` until we role out events.

davemcorwin reviewed (commented):
Thanks @apburnes . Some thoughts:
1. I don't think we need to call out that we're "soft" deleting the site, this is what "delete" means in the context of the system, it's no different than when a user deletes a site. I think this also applies to noting that the infrastructure will be torn down, I'm not sure we need to call it out. Maybe the statement could be more generic and say something like "this remove the site and all associated infrastructure for all users".

2. If we want/need to see "deleted" sites, then maybe we should have an option on the Sites page to show them, since the query itself has to explicitly ask for them. I think when it is deleted, we should redirect to the sites page.

3. The delete confirmation modal is SLICK, but I think forcing the user to enter the repo name is overkill, maybe just a simple confirmation?

4. This failed to delete the site for me locally, but looks like there was an error specific to the local env.

additional comments below

apburnes commented:
Cleaned up the site delete flow but won't work locally unless you comment out the delete infrastructure (Maybe work on local dev feature flag 🤷).  Also simplified the delete verification with just a click.
